iac-extensions/arm/src/main/java/org/sonar/iac/arm/visitors/ArmSymbolVisitor.java

```diff
@@ -96,12 +106,71 @@ void visitAccessUsage(HasIdentifier tree) {
 
     Expression identifier = tree.identifier();
     if (identifier instanceof Identifier identifierTree) {
-      var symbol = currentSymbolTable.getSymbol(identifierTree.value().toLowerCase(Locale.ROOT));
-      if (symbol != null) {
-        symbol.addUsage(tree, Usage.Kind.ACCESS);
-      }
+      addAccessUsageIfSymbolExists(tree, identifierTree.value());
     } else {
       currentSymbolTable.foundUnresolvableSymbolAccess(tree);
     }
   }
+
+  private void addAccessUsageIfSymbolExists(ArmTree tree, String symbolName) {
+    var symbol = currentSymbolTable.getSymbol(symbolName.toLowerCase(Locale.ROOT));
+    if (symbol != null) {
+      symbol.addUsage(tree, Usage.Kind.ACCESS);
+    }
+  }
+
+  private void visitIdentifierJson(IdentifierImpl identifier) {
+    if (!containedInUserAssignedIdentitiesTreeStructure(identifier)) {
+      return;
+    }
+    String variableName = matchesAssignedIdentitiesPattern(identifier.value());
+    if (variableName != null) {
+      addAccessUsageIfSymbolExists(identifier, variableName);
+    }
+  }
+
+  /**
+   * Checks whether the identifier is contained in the following tree structure:
```

Comment:
The comment explains WHAT the method is doing (what is ok in this case), but I need to understand WHY.
It looks like some special case.
Please clarify here in a comment (or in `visitIdentifierJson()`) why we should not count it as usage. 
Also, the ticket & RSPEC say nothing about this case, please update at least the ticket.

Reply:
Ok it is FP you found today, but still the WHY should be documented

Reply:
I will add a comment. Yes it's the FP, I don't think we need to update the rspec for it.